src/js/selfVoicing.js

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2015 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.registerNamespace("gpii.firstDiscovery");
+
+    fluid.defaults("gpii.firstDiscovery.selfVoicing", {
+        gradeNames: ["gpii.firstDiscovery.attachTooltip", "fluid.textToSpeech", "autoInit"],
+        selectors: {
+            mute: ".gpiic-fd-selfVoicing-mute",
+            muteLabel: ".gpiic-fd-selfVoicing-muteLabel"
+        },
+        strings: {
+            muted: "turn voice ON",
+            unmuted: "turn voice OFF"
+        },
+        styles: {
+            muted: "gpii-fd-selfVoicing-muted",
+            unmuted: "gpii-fd-selfVoicing-unmuted"
+        },
+        model: {
+            enabled: false
+        },
+        tooltipContentMap: {
+            "mute": "muted"
+        },
+        invokers: {
+            queueSpeech: {
+                funcName: "gpii.firstDiscovery.selfVoicing.queueSpeech",
+                args: ["{that}", "{arguments}.0", "{arguments}.1"]
+            },
+            toggleState: {
+                funcName: "gpii.firstDiscovery.selfVoicing.toggleState",
+                args: ["{that}"]
+            },
+            setLabel: {
+                funcName: "gpii.firstDiscovery.selfVoicing.setLabel",
+                args: ["{that}.dom.muteLabel", "{that}.options.strings", "{that}.model.enabled"]
+            },
+            setTooltip: {
+                funcName: "gpii.firstDiscovery.selfVoicing.setTooltip",
+                args: ["{that}", "{that}.model.enabled"]
+            },
+            setMuteStyle: {
+                funcName: "gpii.firstDiscovery.selfVoicing.setMuteStyle",
+                args: ["{that}.container", "{that}.options.styles", "{that}.model.enabled"]
+            }
+        },
+        listeners: {
+            "onCreate.bindMute": {
+                "this": "{that}.dom.mute",
+                "method": "click",
+                "args": ["{that}.toggleState"]
+            },
+            "onCreate.setLabel": "{that}.setLabel",
```

Comment:
Probably add a comment here that these listeners can be removed when we start to use model relay.